### PR TITLE
Update React to Version 16

### PR DESCRIPTION
**Pull request by dmytro-danylov**
### 🔦 Summary
We're updating React to v16 in the create-yoshi-app templates:

- fullstack
- fullstack-typescript
- universal
- universal-typescript
- client
- client-typescript
- component-library
- component-library-typescript

We've updated/added the following packages:
react
react-dom
enzyme
enzyme-adapter-react-16

We removed the prop-types package from the typescript templates.

### 🗺️ Test Plan
We've tested if the React 16 works with the current templates, manually and also, running the `npm run test:templates` script.

### CODE REVIEW

**Comment by dmytro-danylov** [comment]
> @dmytro-danylov add a label (Feature Request) so that it will pass all checks

I can't add that label. Looks like I don't have permissions to edit labels.

**Comment by ranyitz** [comment]
@itaibenda @dmytro-danylov When we [release a new version](https://github.com/wix/yoshi/blob/master/CONTRIBUTING.md#release-a-new-version), we use those tags to automatically generate a change and we want one of the Yoshi team collaborators to do it. It's just part of the PR review.

Maybe we need to comunicate it better, so contributors would understand that this is not their fault that this check is not passing.

@Schniz, Maybe the app can say something like "Wait for one of the collaborators to review and add a label" or even better, I think it should be configurable for each project. Similarly to Stale bot's https://github.com/wix/yoshi/blob/master/.github/stale.yml

File: lerna.json [hunk 1]
```diff
@@ -24,5 +24,5 @@
       ]
     }
   },
-  "version": "3.25.1"
```
Comment (ronami) [inline lerna.json]: I think you accidentally changed this, it shouldn't change in PRs (we bump in when releasing a new version).

File: packages/babel-preset-yoshi/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "babel-preset-yoshi",
-  "version": "3.17.0",
```
Comment (ronami) [inline packages/babel-preset-yoshi/package.json]: Same as above.

File: packages/bootstrap-hot-loader/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "bootstrap-hot-loader",
-  "version": "3.19.3",
```
Comment (ronami) [inline packages/bootstrap-hot-loader/package.json]: Same as above.

File: packages/create-yoshi-app/package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "create-yoshi-app",
-  "version": "3.25.0",
```
Comment (ronami) [inline packages/create-yoshi-app/package.json]: Same as above.

File: packages/create-yoshi-app/templates/fullstack/{%packagejson%} [hunk 5]
```diff
@@ -64,6 +65,11 @@
   },
   "eslintConfig": {
     "extends": "yoshi",
+    "settings": {
```
Comment (ronami) [inline packages/create-yoshi-app/templates/fullstack/{%packagejson%}]: Can we move this to `eslint-config-yoshi`? Similar to how it's done in [create-react-app](https://github.com/facebook/create-react-app/blob/master/packages/eslint-config-react-app/index.js#L49).

File: packages/create-yoshi-app/templates/universal/{%packagejson%} [hunk 6]
```diff
@@ -66,6 +67,11 @@
     ]
   },
   "eslintConfig": {
-    "extends": "yoshi"
+    "extends": "yoshi",
+    "settings": {
```
Comment (ronami) [inline packages/create-yoshi-app/templates/universal/{%packagejson%}]: Same as above.

File: packages/eslint-config-yoshi-base/package.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "eslint-config-yoshi-base",
-  "version": "3.25.1",
+  "version": "3.28.0",
```
Comment (ronami) [inline packages/eslint-config-yoshi-base/package.json]: Same as above.

File: packages/eslint-config-yoshi/package.json [hunk 8]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "eslint-config-yoshi",
-  "version": "3.25.1",
+  "version": "3.28.0",
```
Comment (ronami) [inline packages/eslint-config-yoshi/package.json]: Same as above.

File: packages/create-yoshi-app/templates/fullstack-typescript/{%packagejson%} [hunk 9]
```diff
@@ -32,8 +31,9 @@
     "@wix/wix-express-require-https": "latest"
   },
   "devDependencies": {
-    "react-test-renderer": "~15.6.0",
-    "enzyme": "~2.9.0",
+    "react-test-renderer": "~16.7.0",
```
Comment (ronami) [inline packages/create-yoshi-app/templates/fullstack-typescript/{%packagejson%}]: Why do we need `react-test-renderer`? Can we remove it?

**Comment by Schniz** [comment]
> Maybe the app can say something like "Wait for one of the collaborators to review and add a label" or even better, I think it should be configurable for each project

I agree - we can do that. I wanted to use a configuration file to know which labels are valid -- instead of just have the `PR: ` prefix (maybe even reuse `lerna.json`)

The only problem is security teams which can hate the ability to read a file from the repo 😢

**Comment by itaibenda** [comment]
@ronami can we merge this?

**Comment by ronami** [comment]
Closing in favor of #1096.

**Pull request closed without merging**